### DISCUSSION TITLE
open editor to a specific file based on a URL parameter?

### DISCUSSION THREAD

**Discussion by cwinters8** [gitpod-io/openvscode-server#441] [Q&A]
Hey all,

Apologies if this has already been asked - I did search the discussions first but didn't turn up much.

So I know you can open to a specific folder with a URL like this:

```url
https://someservername/?folder=/home/workspace/lessons/0-hello
```

I'd like to be able to add something similar to `&file=main.go` at the end of the URL to open a specific file.

I'm struggling to find documentation on what the available URL parameters are, so if anyone can point me in that direction, I'd be grateful.

Thanks!

**Comment by alexkeizer** [comment]
Did you ever figure it out? I'm looking to do the same thing.

Also, is there a way to make the `?folder=...` parameter work with github links. I tried the following, but neither opened the folder.
```
https://gitpod.io?folder=foo#https://www.github.com/foo/bar.git
https://gitpod.io#https://www.github.com/foo/bar.git?folder=foo
```

**Reply by cwinters8** [reply]
No, I couldn't find a solution to this, and ended up deciding not to use openvscode-server as the editor solution for the project I was working on.

**Reply by jeanp413** [reply]
@alexkeizer if you are using gitpod and you want to open a specific folder within the repo then you need to configure it in the `.gitpod.yml` file https://www.gitpod.io/docs/references/gitpod-yml#workspacelocation

**Reply by alexkeizer** [reply]
Thanks for the hint, but that's not quite what I was looking for. I want to publicly share a preconfigured repository to showcase my project, but the workspace contains a bunch of project configuration files. Those are important and should be opened by vscode, but I don't want the users (that might be unfamiliar with these configuration files) to be confused as to which file is interesting.

In the end I figured it out, specifically if we want to share a file from a github repository! 
Rather than prefixing the repository link with `gitpod.io`, we can also take a link directly to a file.
```
https://gitpod.io#https://github.com/alexkeizer/qpf4-example/blob/main/Scratch/Examples.lean
```
Then, the workspace automatically opens the `Scratch/Examples.lean` file!
